### PR TITLE
Fix Bug 1138117 - Translation Bar comes up even when the page is offered in user's secondary accept language

### PR DESCRIPTION
**Pull request by kyoshino**

### CODE REVIEW

**Comment by alexgibson** [comment]
Thanks @kyoshino - these code changes look fine. Just one question regarding this change:

If I'm reading things correctly, this now means the transbar will now never show if the page locale matches any of the users languages, right? So, for example if I had `['de', 'en-US']` in my accept-language header and visited an `en-US` page, the transbar now wouldn't show, correct?

This does make sense (i think), since the user has explicitly included the page locale in their list of languages. But just wanted to double check if this is indeed intended.

**Comment by pascalchevrel** [comment]
I am not sure I understand the patch, but if my accept-lang headers are 'fr, en-US' and I visit an en-US page, I should be proposed the French version because this is the experience I expect as a user. English is also a fallback locale for almost all of our locales (http://transvision.mozfr.org/string/?entity=toolkit/chrome/global/intl.properties:intl.accept_languages&repo=aurora), that means that people would never be proposed the page in their language if they visit a page in English.

**Comment by alexgibson** [comment]
Tested this locally and I can confirm if I have my accept languages as `['de', 'en']` and visit an `en-US` page, then I don't see the transbar with this change.

**Comment by kyoshino** [comment]
Um, you are right @alexgibson. The priority of the accept languages should be checked... Will update PR later.

**Comment by kyoshino** [comment]
Okay, now I get the right results here. @alexgibson r?

**Comment by alexgibson** [comment]
Thanks @kyoshino - will try and look at this soon :+1:

File: bedrock/tabzilla/templates/tabzilla/tabzilla.js [hunk 1]
```diff
@@ -388,73 +388,81 @@ var Tabzilla = (function (Tabzilla) {
...
+        // is a language, the value is the relevant <link> or <option> element
+        if ($links.length) {
+            $links.each(function () {
+                availableLangs[normalize(this.hreflang)] = this;
+            });
+        } else if ($options.length) {
+            $options.each(function () {
+                availableLangs[normalize(this.value.match(/^\/?([\w\-]+)/)[1])] = this;
+            });
+        }
```
Comment (alexgibson) [inline bedrock/tabzilla/templates/tabzilla/tabzilla.js]: If both `$links.length` and `$options.length` are `0` here, could just `return false;` early and avoid iterating through the user languages below?

File: bedrock/tabzilla/templates/tabzilla/tabzilla.js [hunk 2]
```diff
@@ -388,73 +388,81 @@ var Tabzilla = (function (Tabzilla) {
             return false;
         }
 
+        var normalize = function (lang) {
+            return lang.replace(/^(\w+)(?:-(\w+))?$/, function (m, p1, p2) {
+                return p1.toLowerCase() + ((p2) ? '-' + p2.toUpperCase() : '');
+            });
+        };
+
+        // Normalize the every language for easier comparison
```
Comment (alexgibson) [inline bedrock/tabzilla/templates/tabzilla/tabzilla.js]: Nit: remove `the` in the comment here

File: bedrock/tabzilla/templates/tabzilla/tabzilla.js [hunk 3]
```diff
@@ -388,73 +388,81 @@ var Tabzilla = (function (Tabzilla) {
             return false;
         }
 
+        var normalize = function (lang) {
```
Comment (alexgibson) [inline bedrock/tabzilla/templates/tabzilla/tabzilla.js]: Worth preserving the comment here? `// Normalize the user language in the form of ab or ab-CD`

**Comment by alexgibson** [comment]
I've tested this locally and the revised logic seems sound. All existing tests still pass as well as the new ones added in this PR. 

r+wc from me thanks @kyoshino  :+1:

**Comment by kyoshino** [comment]
Thank you @alexgibson, I have fixed the rest of the issues.

**Pull request merged**